include/GoogleSync/GoogleSync.php

```diff
@@ -0,0 +1,1167 @@
+            }
+        }
+
+        // Now, we look at the Google Calendar
+        $googleEvents = $this->getUserGoogleEvents();
+        if (!isset($googleEvents)) {
+            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Unable to get Google Events');
+            return false;
+        }
+
+        foreach ($googleEvents as $gevent) {
+            if (!$meeting = $this->getMeetingByEventId($gevent->getId())) {
+                $meeting = null;
+            }
+
+            $dowhat = $this->pushPullSkip($gevent, $meeting);
+
+            switch ($dowhat) {
+                case "push":
+                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Pushing Event: ' . $gevent->getSummary());
+                    $this->pushEvent($meeting, $gevent);
+                    break;
+                case "pull":
+                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Pulling Event: ' . $gevent->getSummary());
+                    $this->pullEvent($gevent, $meeting);
+                    break;
+                case "skip":
+                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Skipping Event: ' . $gevent->getSummary());
+                    break;
+                case "push_delete":
+                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Deleting Event: ' . $meeting->name);
+                    $this->delEvent($gevent, $meeting->id);
+                    break;
+                case "pull_delete":
+                    $GLOBALS['log']->info(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Deleting Meeting: ' . $meeting->name);
+                    $this->delMeeting($meeting);
+                    break;
+                default:
+                    $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'pushPullSkip() returned unknown value: ' . $dowhat . ' for event: ' . $gevent->getSummary());
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Setup array of users to sync
+     *
+     * Fills the $users array with users that are configured to sync
+     *
+     * @return null
+     */
+    public function setSyncUsers()
+    {
+
+        $query = "SELECT id FROM users WHERE deleted = '0'";
+        $result = $this->db->query($query);
+
+        require_once('modules/Users/User.php');
+        while ($row = $this->db->fetchByAssoc($result)) {
+            $user = new User();
+            $user->retrieve($row['id']);
+
+            if (!empty($user->getPreference('GoogleApiToken', 'GoogleSync')) &&
+            $accessToken = json_decode(base64_decode($user->getPreference('GoogleApiToken', 'GoogleSync'))) &&
+            $user->getPreference('syncGCal', 'GoogleSync') == '1') {
+                $this->addUser($user->id, $user->full_name);
```

Comment:
if you pass back the results of addUser method as a return array, it makes your function more sense.